File: Data Structures & Algorithms/unboundedKnapsack/submission-2.cpp
class Solution {
public:
    int maximumProfit(vector<int>& val, vector<int>& wt, int cap) {
        int n = wt.size();
        vector<int> prev(cap + 1, 0);
        vector<int> cur(cap + 1, 0);

        for(int j = 0; j <= cap; j++) {
            if(wt[0] <= j) {
                int fits = j/wt[0];
                prev[j] = val[0] * fits;
            }
        }

        for(int i = 1; i < n; i++) {
            for(int j = 1; j <= cap; j++) {
                int skip = prev[j];
                int take = 0;

                if(wt[i] <= j) {
                    take = val[i] + cur[j - wt[i]];
                }

                cur[j] = max(skip, take);
            }
            prev = cur;
        }

        return prev[cap];
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [0] * (n+2)
        for i in range(2, n+2):
            skip = dp[i-1]
            take = nums[i-2] + dp[i-2]
            dp[i] = max(skip, take)
        return dp[n+1]